File: common/exception/src/main/java/org/yechan/Status.java
package org.yechan;

public enum Status {
    BAD_REQUEST,
    CONFLICT,
    RESOURCE_NOT_FOUND,
    INTERNAL_SERVER_ERROR,
    AUTHENTICATION_FAILED,
    ACCESS_DENIED,
    TOKEN_EXPIRED,
    FORBIDDEN;

    public int toHttpStatus() {
        return switch (this) {
            case BAD_REQUEST -> 400;
            case CONFLICT -> 409;
            case RESOURCE_NOT_FOUND -> 404;
            case AUTHENTICATION_FAILED, TOKEN_EXPIRED -> 401;
            case ACCESS_DENIED, FORBIDDEN -> 403;
            case INTERNAL_SERVER_ERROR -> 500;
        };
    }
}

File: common/exception/src/main/kotlin/org/yechan/Status.kt
package org.yechan

enum class Status {
    BAD_REQUEST,
    CONFLICT,
    RESOURCE_NOT_FOUND,
    INTERNAL_SERVER_ERROR,
    AUTHENTICATION_FAILED,
    ACCESS_DENIED,
    TOKEN_EXPIRED,
    FORBIDDEN,
    ;

    fun toHttpStatus(): Int = when (this) {
        BAD_REQUEST -> 400
        CONFLICT -> 409
        RESOURCE_NOT_FOUND -> 404
        AUTHENTICATION_FAILED -> 401
        ACCESS_DENIED -> 403
        TOKEN_EXPIRED -> 401
        INTERNAL_SERVER_ERROR -> 500
        FORBIDDEN -> 403
    }
}
